onnx/defs/tensor/defs.cc

```diff
@@ -685,9 +685,14 @@ ONNX_OPERATOR_SET_SCHEMA(
             return;
           }
 
-          int r = ctx.getInputType(0)->tensor_type().shape().dim_size();
-          int q = ctx.getInputType(1)->tensor_type().shape().dim_size();
-
+          auto& data_shape = ctx.getInputType(0)->tensor_type().shape();
+          auto& indices_shape = ctx.getInputType(1)->tensor_type().shape();
+          int r = data_shape.dim_size();
+          int q = indices_shape.dim_size();
+          int axis = static_cast<int>(getAttribute(ctx, "axis", 0));
```

Comment:
Is this cast safe? why?

Reply:
check if axis is in [-r, r-1]?

Reply:
Fixed.

Reply:
The cast is safe because it is from `int64_t` to `int`